include/zephyr/sys/util.h

```diff
@@ -259,6 +259,22 @@ extern "C" {
  */
 #define DIV_ROUND_UP(n, d) (((n) + (d) - 1) / (d))
 
+/**
+ * @brief Divide and round to the nearest integer.
+ *
+ * Example:
+ * @code{.c}
+ * DIV_ROUND_NEAREST(5, 2); // 3
+ * DIV_ROUND_NEAREST(5, 3); // 2
+ * @endcode
+ *
+ * @param n Numerator.
+ * @param d Denominator.
+ *
+ * @return The result of @p n / @p d, rounded to the nearest integer.
+ */
+#define DIV_ROUND_NEAREST(n, d) (((n) + ((d) / 2)) / (d))
```

Comment:
Macro hygiene pedantry: this expands the `d` argument twice.  That's not illegal (and I see DIV_ROUND_UP already does the same thing) but it's sort of a footgun and would be good to call out in the docs; maybe as a separate patch that adds the same warning elsewhere.

A better trick would be to use a temporary in a statement expression (which IIRC are allowed now?) or to define it as an ALWAYS_INLINE function, of course.

Reply:
Interesting... Linux does it in a funky way too https://elixir.bootlin.com/linux/latest/source/include/linux/math.h#L86

Btw should we match the Linux name? `DIV_ROUND_CLOSEST`

Reply:
After looking into the Linux code that Fabio posted I realized that my logic only worked with positive integers.
Changes to the latest version:

1. Changed the name to `DIV_ROUND_CLOSEST` to match linux.
2. Converted the macro to an ALWAYS_INLINE function, int is used for data type.
3. Added support for rounding negative integers and updated the test.

Please let me know what you think.

BTW it's quite interesting that implementing a rounding helper turned out to be rather non-trivial task :)